helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/metrics/model/Metric.java

```diff
@@ -0,0 +1,54 @@
+package org.apache.helix.controller.rebalancer.waged.metrics.model;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.helix.monitoring.mbeans.dynamicMBeans.DynamicMetric;
+
+/**
+ * Defines a generic metric interface.
+ */
+public interface Metric {
+
+  /**
+   * Gets the name of the metric.
+   */
+  String getMetricName();
+
+  /**
+   * Resets the internal state of this metric.
+   */
+  void reset();
+
+  /**
+   * Prints the metric along with its name.
+   */
+  String print();
+
+  /**
+   * Returns an appropriate value for the metric at the time of the call.
+   * @return metric value
+   */
+  long getMetricValue();
```

Comment:
It seems that this method is only for testing, right? I have some concern, the definition is not crystal clear. What is "appropriate"? What's the expected return for the histogram, or a counter? They are all different. For the user of the interface, it would hard to determine how to use it.